### PR TITLE
EXPDAT-012/002 Maps with current location on the HomeFragment

### PR DESCRIPTION
**Pull request by bengaal**
Add Maps to home fragment showing current position.

### CODE REVIEW

**Comment by steschi** [comment]
any clue why the github build system says it does not compile?

also: yes we agreed to commit .idea/ folder files but only useful ones. I am not sure why changing the JavaSDK from 1.8 to 1.11 should be in this pull request.

**Comment by bengaal** [comment]
> any clue why the github build system says it does not compile?
> 
> also: yes we agreed to commit .idea/ folder files but only useful ones. I am not sure why changing the JavaSDK from 1.8 to 1.11 should be in this pull request.

It doesn't compile because the file, the secret API key was in is missing. I am going to fix it as soon as I have some time. As we discussed it the last time, for some reason Android Studio forces Java 1.11 for me. I am going to take a look at it. 
Edit: Whenever I save the `languageLevel` jumps back to JDK_11, by the way, why are we using a legacy version @steschi?
Also maybe you or @benjowun could check if for you guys it is possible to change it to JDK_1_8
Edit2: This parameter doesn't seem to actually change the used JDK version.

**Comment by steschi** [comment]
The Java version seems to reflect the Java used for the IDE (Android Studio). All windows users use the installer of Android Studio which is delivered with its own Java 1.8 runtime. I am not sure why your IDE is using 1.11 though.

**Comment by steschi** [comment]
as a reviewer i could delete the changes made for the Java Version issue and then approve the PR, i am just not sure if it is the correct way based on the course rules.

File: app/src/androidTest/java/at/sw21_tug/team_25/expirydates/HomeFragmentMapTest.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @Rule
+    @JvmField
+    var mGrantPermissionRule =
+            GrantPermissionRule.grant(
+                    "android.permission.ACCESS_FINE_LOCATION")
+
+    @Test
+    fun homeFragmentMapTest() {
+        val view = onView(
+allOf(withContentDescription("Google Map"),
```
Comment (steschi) [inline app/src/androidTest/java/at/sw21_tug/team_25/expirydates/HomeFragmentMapTest.kt]: could you please format the file? I know it was generated. Code->Reformat File ....

File: app/src/main/java/at/sw21_tug/team_25/expirydates/MainActivity.kt [hunk 2]
```diff
@@ -66,8 +75,48 @@ class MainActivity : AppCompatActivity() {
...
+            } else {
+                ActivityCompat.requestPermissions(this,
+                    arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), REQUEST_LOCATION)
+            }
+        }
+    }
+
+
+
+//    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>,
```
Comment (steschi) [inline app/src/main/java/at/sw21_tug/team_25/expirydates/MainActivity.kt]: why is this in here? do we need it later on?

File: app/src/main/java/at/sw21_tug/team_25/expirydates/ui/home/HomeFragment.kt [hunk 3]
```diff
@@ -1,23 +1,50 @@
 package at.sw21_tug.team_25.expirydates.ui.home
 
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.AlertDialog
```
Comment (steschi) [inline app/src/main/java/at/sw21_tug/team_25/expirydates/ui/home/HomeFragment.kt]: Please remove all unused imports

**Review by steschi — CHANGES_REQUESTED** [review]

**Comment by bengaal** [comment]
> The Java version seems to reflect the Java used for the IDE (Android Studio). All windows users use the installer of Android Studio which is delivered with its own Java 1.8 runtime. I am not sure why your IDE is using 1.11 though.

Me neither I am using the most recent build from the snapstore on Ubuntu.

File: app/src/androidTest/java/at/sw21_tug/team_25/expirydates/HomeFragmentMapTest.kt (hunk 1 above)
Comment (bengaal) [reply app/src/androidTest/java/at/sw21_tug/team_25/expirydates/HomeFragmentMapTest.kt]: Yeah sure, good point!

File: app/src/main/java/at/sw21_tug/team_25/expirydates/MainActivity.kt (hunk 2 above)
Comment (bengaal) [reply app/src/main/java/at/sw21_tug/team_25/expirydates/MainActivity.kt]: It's just a forgotten leftover, thanks for pointing it out :)

File: app/src/main/java/at/sw21_tug/team_25/expirydates/ui/home/HomeFragment.kt (hunk 3 above)
Comment (bengaal) [reply app/src/main/java/at/sw21_tug/team_25/expirydates/ui/home/HomeFragment.kt]: Sure do!

**Review by steschi — APPROVED** [review]
all testcases pass, app builds and works

**Pull request merged**